### PR TITLE
Print more positions when a race ends

### PR DESCRIPTION
**Pull request by TheMolkaPL**

### CODE REVIEW

File: src/main/java/com/github/hornta/race/RacingManager.java [hunk 1]
```diff
@@ -180,16 +180,26 @@ void onChangeRaceName(RaceChangeNameEvent event) {
 
   @EventHandler
   void onRaceSessionResult(RaceSessionResultEvent event) {
+    List<PlayerSessionResult> sortedResults = new ArrayList<>(128);
```
Comment (hornta) [inline src/main/java/com/github/hornta/race/RacingManager.java:183]: Is 128 needed here? ArrayList grows automatically.

File: src/main/resources/translations/Racing/english.yml [hunk 2]
```diff
@@ -140,7 +140,7 @@ race_start_gamemode_disqualified_target: "§eYou were disqualified since you are
 race_death_disqualified: "§e<player_name> died during the race and is now disqualified."
 race_death_disqualified_target: "§eYou died and is therefore disqualified. Better luck next time."
 edit_no_edit_mode: "§cYou can't change race `<race_name>` because it's not under construction. Type §7/rc setstate <race_name> under_construction§c to put it under construction."
-race_win: "§e<player_name> has won race §a<race_name>§e with time <time>."
+race_win: "§e#<position> <player_name> has finished race §a<race_name>§e with time <time>."
```
Comment (hornta) [inline src/main/resources/translations/Racing/english.yml:143]:
```suggestion
race_participant_result: "#<position>. §e<player_name> finished race §a<race_name>§e with time <time>."
```

Comment (hornta) [reply src/main/resources/translations/Racing/english.yml:143]: Can you rename race_win to something more suiting like race_participant_result or something.

**Comment by hornta** [comment]
Thanks. This is some nice changes. I had some comments for you to view before merging.

File: src/main/java/com/github/hornta/race/RacingManager.java [hunk 3]
```diff
@@ -180,16 +180,26 @@ void onChangeRaceName(RaceChangeNameEvent event) {
...
-        MessageManager.setValue("player_name", entry.getValue().getPlayerSession().getPlayerName());
+      PlayerSessionResult value = entry.getValue();
+      event.getResult().getRaceSession().getRace().addResult(value);
+      sortedResults.add(value);
+    }
+
+    sortedResults.sort(Comparator.comparingInt(PlayerSessionResult::getPosition));
+    for (PlayerSessionResult result : sortedResults) {
+      int position = result.getPosition();
+      if (position >= 1 && position <= 5) {
```
Comment (nordlead2005) [inline src/main/java/com/github/hornta/race/RacingManager.java:193]: You previously were reserving 128 entries, but only print 5 of them here. The Top commands print 10 each. I think this should either match the 10 from the other commands and/or be configurable. I just did a race with 6 people over the weekend and my course holds 8, and 10 isn't an unreasonable amount to print.

File: src/main/resources/translations/Racing/english.yml (hunk 2 above)
Comment (hornta) [reply src/main/resources/translations/Racing/english.yml]: Can we have the player get the §e color? Maybe a different color for the position number?

File: src/main/java/com/github/hornta/race/RacingManager.java (hunk 3 above)
Comment (nordlead2005) [reply src/main/java/com/github/hornta/race/RacingManager.java:193]: position >= 1 is a useless check, as position can't be 0 or less. If there is concern, then there should be a check that prints an error to the log.

File: src/main/java/com/github/hornta/race/RacingManager.java [hunk 4]
```diff
@@ -180,16 +180,26 @@ void onChangeRaceName(RaceChangeNameEvent event) {
...
+    List<PlayerSessionResult> sortedResults = new ArrayList<>();
     for (Map.Entry<RacePlayerSession, PlayerSessionResult> entry : event.getResult().getPlayerResults().entrySet()) {
-      event.getResult().getRaceSession().getRace().addResult(entry.getValue());
-      if (entry.getValue().getPosition() == 1) {
-        MessageManager.setValue("player_name", entry.getValue().getPlayerSession().getPlayerName());
+      PlayerSessionResult value = entry.getValue();
+      event.getResult().getRaceSession().getRace().addResult(value);
+      sortedResults.add(value);
+    }
+
```
Comment (nordlead2005) [inline src/main/java/com/github/hornta/race/RacingManager.java:189]: I feel like there should be a broadcast here that says something like `~~~ <race_name> Results - <lap> Lap<s> ~~~` before printing out the actual results. Much like the top/info commands have a header before dumping their information.

**Comment by hornta** [comment]
LGTM

File: src/main/java/com/github/hornta/race/RacingManager.java (hunk 4 above)
Comment (hornta) [reply src/main/java/com/github/hornta/race/RacingManager.java]: I'm all for it! 👍

**Comment by TheMolkaPL** [comment]
Sorry about the delay, but my PSU was broken and I'm fixing it currently. I was going to add the header, but I don't have second machine to program on.

**Pull request merged**